### PR TITLE
EDU-3694: Workflow Update

### PR DESCRIPTION
**Pull request by stephanos**
## What does this PR do?

Touch up on the Workflow Update documentation for its GA release.

- [x] improve encyclopedia docs
- [x] improve Go SDK docs

### CODE REVIEW

**Comment by fairlydurable** [comment]
Hi @stephanos. Is there an EDU ticket for this PR work? If not, can you file one please? Thanks!

File: docs/encyclopedia/application-message-passing.mdx [hunk 1]
```diff
@@ -168,6 +170,8 @@ The start call will give you a handle you can use to track the Update, determine
 
 If you want to send an Update to another Workflow such as a Child Workflow from within a Workflow, you should do so within an Activity and use the Temporal Client as normal.
 
+Use [Update Validators](#update-validators) and [Update IDs](#exactly-once-message-processing) to stay within the system limits in both [Cloud](/cloud/limits#per-workflow-execution-update-limits) and [Self-Hosted](/self-hosted-guide/defaults).
```
Comment (stephanos) [inline docs/encyclopedia/application-message-passing.mdx:173]: @drewhoskins-temporal @dandavison  what do you think about this?

File: docs/develop/go/message-passing.mdx [hunk 2]
```diff
@@ -346,10 +346,24 @@ You must provide the Workflow Id, but specifying a Run Id is optional.
...
+- If `WaitForStage` is set to `WorkflowUpdateStageCompleted`, the handle is returned after the Update completes.
+- If `WaitForStage` is set to `WorkflowUpdateStageAccepted`, the handle is returned after the Update is accepted and the Validator is run, if it's defined.
+
+Note: You cannot send Updates to other Workflow Executions.
+
+The following sample demonstrates how to create a context with timeout for sending an `Update` to the Workflow:
+
+If `WaitForStage` is set to `WorkflowUpdateStageCompleted` the handle is returned after the update completes; if `WaitForStage` is set to `WorkflowUpdateStageAccepted` the handle is returned after the Update is accepted (i.e. after the validator has run, if there is a validator).
+
+Also note that you can't send Updates to other Workflow Executions.
```
Comment (stephanos) [inline docs/develop/go/message-passing.mdx:360]: I believe Erica added these to align Go with the other SDKs; as I can see similar copy over there.

File: docs/develop/go/message-passing.mdx [hunk 3]
```diff
@@ -334,8 +334,8 @@ An Update is a synchronous, blocking call that can change Workflow state, contro
 
 A Client sending an Update must wait until the Server delivers the Update to a Worker.
 Workers must be available and responsive.
+Setting a timeout with a context provides a hard limit on how long a client will wait for a response.
```
Comment (stephanos) [inline docs/develop/go/message-passing.mdx:337]: This was my addition to make sure users think about setting a timeout.

Comment (stephanos) [reply docs/develop/go/message-passing.mdx:337]: I'm not super experienced in the other SDKs to know where/how to put that there, too. Anyway, something we can always do later.

File: docs/develop/go/message-passing.mdx [hunk 4]
```diff
@@ -346,10 +346,24 @@ You must provide the Workflow Id, but specifying a Run Id is optional.
 If you supply only the Workflow Id (and provide an empty string as the Run Id param), the running Workflow Execution receives the Update.
 
 You must provide a `WaitForStage` when calling `UpdateWorkflow()`.
-This parameter controls what stage the update must reach before a handle is returned to the caller. If `WaitForStage` is set to `WorkflowUpdateStageCompleted` the handle is returned after the update completes; if `WaitForStage` is set to `WorkflowUpdateStageAccepted` the handle is returned after the Update is accepted (i.e. after the validator has run, if there is a validator).
+This parameter controls the stage the update must reach before returning a handle to the caller:
+
+- If `WaitForStage` is set to `WorkflowUpdateStageCompleted`, the handle is returned after the Update completes.
+- If `WaitForStage` is set to `WorkflowUpdateStageAccepted`, the handle is returned after the Update is accepted and the Validator is run, if it's defined.
```
[the remark this answers is not in the document]
Comment (stephanos) [reply docs/develop/go/message-passing.mdx:352]: Right; maybe a merge was messed up, the bottom part seems to be redundant. I'll merge them.

File: docs/develop/go/message-passing.mdx (hunk 3 above)
Comment (brianmacdonald-temporal) [reply docs/develop/go/message-passing.mdx:337]: I like that; nice addition.

File: docs/encyclopedia/application-message-passing.mdx (hunk 1 above)
Comment (brianmacdonald-temporal) [reply docs/encyclopedia/application-message-passing.mdx:172]: Agreed; nice addition.

**Review by brianmacdonald-temporal — APPROVED** [review]
These changes look good to me.
Do we also want to remove the Stability tip under the Update-With-Start heading?

**Comment by stephanos** [comment]
> Do we also want to remove the Stability tip under the Update-With-Start heading?

I think it's useful to have since UwS will only work as intended on that particular version (even though it's been part of previous versions).

**Pull request merged**